src/main/java/io/appform/functionmetrics/MethodData.java

```diff
@@ -22,10 +22,12 @@
 public class MethodData {
     private final String className;
     private final String methodName;
+    private final boolean parameterCaptureRequired;
```

Comment:
You can actually store a list of parameter indexes that need to be read instead of the boolean. As a matter of fact you can store a commons StrSubstitutor format string here and fill in the params at runtime to generate the metric name in a single shot.
Can also add a isParamCaptureRequired() method to return !list.isEmpty()
Advantages:
- SPEEEEEED

Reply:
Ack. Enabling parameter capture definitely has some overhead. As per actual stats in production, even when parameter capture is enabled, it is done so for less than 5% of the methods which are monitored. So I wanted to avoid this overhead of constructing parameter string for methods that don't require it (close to 95%). I will pick this optimisation as a follow up. I will add a TODO for now.

Reply:
Done.